AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/physx_collider.py

```diff
@@ -0,0 +1,318 @@
+from editor_python_test_tools.asset_utils import Asset
+
+COMPONENT_NAME = "PhysX Collider"
+
+class PhysxCollider:
+    """
+    PhysxCollider class is used to interact with the PhysXCollider Entity Componenet and its properties. This class
+        makes it easy for the user to interact with the PhysxCollider in a natural and easy-to-write/maintain way.
+    The PhysxCollider object stores a reference to an EditorComponenet and is created by passing an EditorEntity to the
+    PhysxCollider constructor via PhysxCollider(EditorEntity).
+    """
+    def __init__(self, editor_entity: EditorEntity) -> None:
+        self.component = editor_entity.add_component(COMPONENT_NAME)
+
+    class Path:
+        """
+        A container class for the PropertyTree paths relevant to the Physx Collider. This tree can be returned by
+            calling self.component.get_property_type_visibility() which will also return the expected data type
+            and visibility.
+        """
+        class Box:
+            BASE = 'Shape Configuration|Box'
+            DIMENSIONS = 'Shape Configuration|Box|Dimensions'
+        class Sphere:
+            BASE = 'Shape Configuration|Sphere'
+            RADIUS = 'Shape Configuration|Sphere|Radius'
+        class Capsule:
+            BASE = 'Shape Configuration|Capsule'
+            HEIGHT = 'Shape Configuration|Capsule|Height'
+            RADIUS = 'Shape Configuration|Capsule|Radius'
+        class Cylinder:
+            BASE = 'Shape Configuration|Cylinder'
+            RADIUS = 'Shape Configuration|Cylinder|Radius'
+            HEIGHT = 'Shape Configuration|Cylinder|Height'
+            SUBDIVISION = 'Shape Configuration|Cylinder|Subdivision'
+        class PhysicsAsset:
+            BASE = 'Shape Configuration|Asset'
+            PHYSX_MESH = 'Shape Configuration|Asset|PhysX Mesh'
+            ASSET_SCALE = 'Shape Configuration|Asset|Configuration|Asset Scale'
+            PHYSICS_MATERIALS_FROM_ASSET = 'Shape Configuration|Asset|Configuration|Physics materials from asset'
+
+        COLLISION_LAYER = 'Collider Configuration|Collision Layer'
+        COLLIDES_WITH = 'Collider Configuration|Collides With'
+        TRIGGER = 'Collider Configuration|Trigger'
+        SIMULATED = 'Collider Configuration|Simulated'
+        IN_SCENE_QUERIES = 'Collider Configuration|In Scene Queries'
+        OFFSET = 'Collider Configuration|Offset'
+        ROTATION = 'Collider Configuration|Rotation'
+        PHYSX_MATERIAL_ASSET = ']|'  # GH-12503 Needs a better property path
+        TAG = 'Collider Configuration|Tag'
+        RESET_OFFSET = 'Collider Configuration|Rest offset'
+        CONTACT_OFFSET = 'Collider Configuration|Contact offset'
+        DRAW_COLLIDER = 'Debug draw settings|Draw collider'
+        SHAPE = 'Shape Configuration|Shape'
+
+    # General Properties
+    #TODO: GHI #12632 - Figure out how to set dropdowns for Collison Layer and Collides With
+
+    def toggle_trigger(self) -> None:
```

Comment:
I think this function name would be more clear as `toggle_is_trigger`. For somebody that doesn't know this is enable/disable option it reads more like a setter, so using `toggle_is_trigger` helps to clarify it.